src/cargo/ops/cargo_package.rs

```diff
@@ -611,6 +631,9 @@ fn tar(
                     .with_context(|| {
                         format!("could not archive source file `{}`", disk_path.display())
                     })?;
+                if !(metadata.is_dir() || metadata.file_type().is_symlink()) {
```

Comment:
I have an impression that Cargo resolves all symlinks and copies over (see https://github.com/rust-lang/cargo/issues/11203#issuecomment-1273840632), and there should be only "file" in `FileContents` I guess. That is to say, this check might be redundant?

To prevent regressions, we might also have an additional test to assert this new crate size info for a package containing symlinks.

Reply:
I added a test with a symlink, and it turns out that _both_ of these checks are indeed redundant, so I removed them. Good catch!